agent/config/config.go

```diff
@@ -248,6 +249,20 @@ func EnvironmentConfig() Config {
 		}
 	}
 
+	dockerStopTimeoutSecondsEnv := os.Getenv("ECS_CONTAINER_STOP_TIMEOUT")
+	var dockerStopTimeoutSeconds uint64
+	if dockerStopTimeoutSecondsEnv == "" {
```

Comment:
You don't need to have this defaulting behavior here; `Merge` will correctly handle merging a zero-valued config value with the default.  Instead, you can just leave off the condition where `dockerStopTimeoutSecondsEnv` is 0 and change it to:

```go
if dockerStopTimeoutSecondsEnv != "" {
	// your new code
}
```